airbyte-persistence/job-persistence/src/test/java/io/airbyte/persistence/job/errorreporter/JobErrorReporterTest.java

```diff
@@ -499,4 +499,20 @@ void testReportSpecJobFailure() {
     Mockito.verifyNoMoreInteractions(jobErrorReportingClient);
   }
 
+  @Test
+  void testReportUnsupportedFailureType() {
+    final FailureReason failureReason = new FailureReason()
+        .withMetadata(new Metadata()
+            .withAdditionalProperty(FROM_TRACE_MESSAGE, true)
+            .withAdditionalProperty(CONNECTOR_COMMAND_KEY, READ_COMMAND))
```

Comment:
this test covers the `read` command, should there also be a test for `check` and `discover`? From what I understood is that using `reportJobFailureReason` is so that it can cover all the different commands in the Airbyte Protocol instead of only `read`

Reply:
Updated tests